pkg/vault/contracts/CompositeLiquidityRouter.sol

```diff
@@ -347,15 +384,20 @@ contract CompositeLiquidityRouter is ICompositeLiquidityRouter, BatchRouterCommo
             IERC4626 wrappedToken = IERC4626(address(erc4626PoolTokens[i]));
             IERC20 underlyingToken = IERC20(_vault.getBufferAsset(wrappedToken));
 
+            if (useWrappedTokens[i] == true) {
+                finalAmountsIn[i] = amountsIn[i];
+                _takeTokenIn(params.sender, wrappedToken, finalAmountsIn[i], params.wethIsEth);
+                continue;
+            }
+
             // If the Vault returns address 0 as underlying, it means that the ERC4626 token buffer was not
             // initialized. Thus, the Router treats it as a non-ERC4626 token.
             if (address(underlyingToken) == address(0)) {
```

Comment:
This shouldn't be accepted anymore.

If the user doesn't specify `useWrappedTokens` it means that they want to wrap. If there's no registered asset, the operation is invalid, so just revert.

In other words: in the `main` branch we're guessing whether to use wrapped or underlying. That should no longer be the case; the user input shall be the truth from now on. If the operation can't be fulfilled as specified, revert instead of guessing.